src/app/context/wallet/BTCWalletProvider.tsx

```diff
@@ -137,25 +143,25 @@ export const BTCWalletProvider = ({ children }: PropsWithChildren) => {
           btcAddress: address,
         });
 
-        addBreadcrumb({
-          level: "info",
-          message: "Connect BTC wallet",
-          data: {
-            network,
-            address,
-            publicKeyNoCoord,
-            walletName: await walletProvider.getWalletProviderName(),
-          },
+        logger.info("BTC wallet connected", {
+          network,
+          userPublicKey: publicKeyNoCoord?.toString() || "",
+          btcAddress: address || "",
```

Comment:
Is it possible htat address and/or publickey is undefined? If so shouldn't this be an error? (We are logging it as BTC wallet connected) (this is replicated across other error refactors as well)